module/VuFind/src/VuFind/Search/Factory/AbstractSolrBackendFactory.php

```diff
@@ -183,6 +192,55 @@ public function __invoke(ContainerInterface $sm, $name, array $options = null)
         return $backend;
     }
 
+    /**
+     * Return an ordered array of configurations to check for index configurations.
+     *
+     * @return string[]
+     */
+    protected function getIndexConfigFallbackPath(): array
+    {
+        return array_unique([$this->searchConfig, $this->mainConfig]);
+    }
+
+    /**
+     * Get an index-related configuration setting.
+     *
+     * @param string $setting    Name of setting
+     * @param mixed  $default    Default value if unset
+     * @param string $configName Name of config to look in
+     * @param bool   $fallback   Should we fall back to main config if the
+     * setting is absent from the specified config file?
+     *
+     * @return mixed
+     */
+    protected function getIndexConfig(
```

Comment:
I find this method a bit confusing. It might be clearer to first determine the paths to check according to `$fallback` and then just loop through the path array.

Before that, however, my concern is that this method potentially does a lot of repeated work for each setting retrieved. How would you feel about merging the config sections the first time and caching the results?